actions/setup.go

```diff
@@ -54,101 +101,182 @@ func DefaultLocalSetup(
 	if err != nil {
 		return nil, err
 	}
-
 	err = env.DeploySpecs(initialDeployInitFunc)
 	if err != nil {
 		return nil, err
 	}
 
-	// Initialize blockchain client
-	var bcc client.BlockchainClient
-	switch network.Config().Type {
-	case client.BlockchainTypeEVMMultinode:
-		bcc, err = environment.NewBlockchainClients(env, network)
-	case client.BlockchainTypeEVM:
-		bcc, err = environment.NewBlockchainClient(env, network)
-	}
+	networkInfo, err := getNetworkInfo(network, env)
 	if err != nil {
 		return nil, err
 	}
 
-	// Initialize wallets
-	wallets, err := network.Wallets()
+	// configure default retry
+	retry.DefaultAttempts = conf.Retry.Attempts
+	retry.DefaultDelayType = func(n uint, err error, config *retry.Config) time.Duration {
+		return conf.Retry.LinearDelay
+	}
+
+	return &DefaultSuiteSetup{
+		Config:   conf,
+		Client:   networkInfo.Client,
+		Wallets:  networkInfo.Wallets,
+		Deployer: networkInfo.Deployer,
+		Link:     networkInfo.Link,
+		Env:      env,
+		Network:  network,
+	}, nil
+}
+
+// TearDown checks for test failure, writes logs if there is one, then tears down the test environment, based on the
+// keep_environments config value
+func (s *DefaultSuiteSetup) TearDown() func() {
+	return teardown(*s.Config, s.Env, s.Client)
+}
+
+// MultiNetworkSuiteSetup enables friendly usage of multiple networks
+type MultiNetworkSuiteSetup interface {
```

Comment:
👍 for wrapping the suite setup in an interface. Could we get the existing single network suite setup to implement this interface (maybe renaming the interface to just `SuiteSetup`) and adding a `DefaultNetwork` function that returns the first network in the map/only network.